git-hg/mercurialToGit.sh

```diff
@@ -74,19 +74,21 @@ fi
 
 # JDK11+ tag sorting:
 # We use sort and tail to choose the latest tag in case more than one refers the same commit.
-# Versions tags are formatted: jdk-V[.W[.X]]+B; with V, W, X, B being numeric.
-# Transform "-" to "." in tag so we can sort as: "jdk.V[.W[.X]]+B"
+# Versions tags are formatted: jdk-V[.W[.X[.P]]]+B; with V, W, X, P, B being numeric.
+# Transform "-" to "." in tag so we can sort as: "jdk.V[.W[.X[.P]]]+B"
+# Transform "+" to ".0.+" during the sort so that .P (patch) is defaulted to "0" for those
+# that don't have one, and the trailing "." to terminate the 5th field from the +
 #
-# Example: jdk-11.0.2+2, jdk-11.0.1+3, jdk-11.0.2+1
-# Sort by "B": jdk-11.0.2+1, jdk-11.0.2+2, jdk-11.0.1+3
-# Then, "Stable" sort by V.W.X: jdk-11.0.1+3, jdk-11.0.2+1, jdk-11.0.2+2
-# Latest = jdk-11.0.2+2
+# Example: jdk-11.0.2+2, jdk-11.0.1+3, jdk-11.0.2+1, jdk-11.0.2.1+2
+# Sort by "B": jdk-11.0.2+1, jdk-11.0.2+2, jdk-11.0.2.1+2, jdk-11.0.1+3,
+# Then, "Stable" sort by V.W.X.P: jdk-11.0.1+3, jdk-11.0.2+1, jdk-11.0.2+2, jdk-11.0.2.1+2
+# Latest = jdk-11.0.2.1+2
 #
 # First, sort on build number (B):
-jdk11plus_tag_sort1="sort -t+ -k2n"
-# Second, (stable) sort on (V), (W), (X):
-jdk11plus_tag_sort2="sort -t. -k2n -k3n -k4n -s"
-jdk11plus_sort_tags_cmd="grep -v _adopt | sed 's/jdk-/jdk./g' | $jdk11plus_tag_sort1 | $jdk11plus_tag_sort2 | sed 's/jdk./jdk-/g'"
+jdk11plus_tag_sort1="sort -t+ -k2,2n"
+# Second, (stable) sort on (V), (W), (X), (P): P(Patch) is optional and defaulted to "0"
+jdk11plus_tag_sort2="sort -t. -k2,2n -k3,3n -k4,4n -k5,5n"
+jdk11plus_sort_tags_cmd="grep -v _openj9 | grep -v _adopt | sed 's/jdk-/jdk./g' | sed 's/+/.0.+/g' | $jdk11plus_tag_sort1 | nl | $jdk11plus_tag_sort2 | sed 's/\.0\.+/+/g' | cut -f2- | sed 's/jdk./jdk-/g'"
```

Comment:
is `cut -f2-` correct?

Reply:
yes, this is copied from sbin/build.sh. Basically, to make the cmd work on Solaris that does not support "stable" sorting, we use the "trick" of line numbering ("nl") after the first sort, then doing sort2, then cut -f2- removes the first field "line number".